src/qonnx/transformation/quantize_graph.py

```diff
@@ -0,0 +1,261 @@
+        scale_value,
+        zeropoint_value,
+        bitwidth_value,
+        narrow,
+        signed,
+        rounding_mode,
+    ):
+        quantnode_output_dtype = DataType["UINT8"]
+        quant_tensor = onnx.helper.make_tensor_value_info(
+            model.make_new_valueinfo_name(), TensorProto.FLOAT, quantnode_output_shape
+        )
+        model.graph.value_info.append(quant_tensor)
+        model.set_tensor_datatype(quant_tensor.name, quantnode_output_dtype)
+
+        stationary_input_dtype = DataType["FLOAT32"]
+        scale_tensor = np.array(scale_value).astype(np.float32)
+        s_value = onnx.helper.make_tensor_value_info(
+            model.make_new_valueinfo_name(), TensorProto.FLOAT, quantnode_output_shape
+        )
+        model.graph.value_info.append(s_value)
+        model.set_tensor_datatype(s_value.name, stationary_input_dtype)
+        model.set_initializer(s_value.name, scale_tensor)
+
+        zeropt_tensor = np.array(zeropoint_value).astype(np.float32)
+        z_value = onnx.helper.make_tensor_value_info(
+            model.make_new_valueinfo_name(), TensorProto.FLOAT, quantnode_output_shape
+        )
+        model.graph.value_info.append(z_value)
+        model.set_tensor_datatype(z_value.name, stationary_input_dtype)
+        model.set_initializer(z_value.name, zeropt_tensor)
+
+        bitwidth_tensor = np.array(bitwidth_value).astype(np.float32)
+        b_value = onnx.helper.make_tensor_value_info(model.make_new_valueinfo_name(), TensorProto.FLOAT, [1])
+        model.graph.value_info.append(b_value)
+        model.set_tensor_datatype(b_value.name, stationary_input_dtype)
+        model.set_initializer(b_value.name, bitwidth_tensor)
+
+        quant_node = onnx.helper.make_node(
+            "Quant",
+            inputs=[quantnode_input, s_value.name, z_value.name, b_value.name],
+            outputs=[quant_tensor.name],
+            name="Quant_node_" + str(node_count) + str(tensor_count),
+            narrow=narrow,
+            signed=signed,
+            rounding_mode=rounding_mode,
+        )
+
+        return quant_node, quant_tensor
+
+    def adjust_graph(self, model, input_positions, node_in_focus, quantized_nodes, node_count):
+        tensor_count = 0
+        for pos in input_positions:
+            node_details = (node_in_focus.name, pos[0])
+            if (
+                node_details not in quantized_nodes
+            ):  # This is to ensure that we don't quantize the same node for the same input/output index.
+                if pos[0][0] == "input":
+                    input_to_quantnode = node_in_focus.input[pos[0][1]]
+                    consumer_node = node_in_focus
+                    producer_node = model.find_producer(input_to_quantnode)
+                    if producer_node is None or producer_node.op_type != "Quant":
+                        quantization_to_perform = "yes"
+                    else:
+                        quantization_to_perform = "no"
```

Comment:
why the strings "yes" and "no"? just use True and False as bool values

Reply:
Suggested changes were performed.